src/gentropy/common/spark_helpers.py

```diff
@@ -375,6 +376,78 @@ def order_array_of_structs_by_field(column_name: str, field_name: str) -> Column
     )
 
 
+def order_array_of_structs_by_two_fields(
+    array_name: str, col1: str, col2: str
+) -> Column:
+    """Sort array of structs by a field in descending order and by an other field in an ascending order.
+
+    This function doesn't deal with null values, assumes the sort columns are not nullable.
+
+    Args:
+        array_name (str): Column name with array of structs
+        col1 (str): Name of the first keys sorted in descending order
+        col2 (str): Name of the second keys sorted in ascending order
+
+    Returns:
+        Column: Sorted column
+    """
+    return f.expr(
+        f"""
+        array_sort(
+        {array_name},
+        (left, right) -> case
+                when left.{col1} is null and right.{col1} is null then 0
+                when left.{col2} is null and right.{col2} is null then 0
+
+                when left.{col1} is null then 1
+                when right.{col1} is null then -1
+
+                when left.{col2} is null then 1
+                when right.{col2} is null then -1
+
+                when left.{col1} < right.{col1} then 1
+                when left.{col1} > right.{col1} then -1
+                when left.{col1} == right.{col1} and left.{col2} > right.{col2} then 1
```

Comment:
This would incline that the ordering is done firstly by the col1, and in case they are equal, then by col2. You do not compare the case when `left.col1 is Null and right.col1 is Null, but left.col2 > right.col2` and others as far as I understand it. It's quiet complex operation.

Reply:
yes, that's why the docstring says:
```
This function doesn't deal with null values, assumes the sort columns are not nullable.
```

I didn't want to make this expression even more elaborate. The method is used when I need to sort object in the vep column and I can be confident these values are always present.